airflow/models.py

```diff
@@ -1801,6 +1801,50 @@ def __init__(self, event, task_instance, owner=None, extra=None, **kwargs):
         self.owner = owner or task_owner
 
 
+class SkipMixin(object):
+    def skip(self, dag_run, execution_date, tasks):
```

Comment:
The non dag run version needs tasks instead of task ids, as tasks are always available and going from task_id to task requires a look up I have chosen for tasks.